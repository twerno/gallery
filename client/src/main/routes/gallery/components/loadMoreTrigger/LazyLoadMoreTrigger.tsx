import AnimatedLoader from 'main/components/AnimatedLoader';
import { LazyLoader } from 'main/components/LazyLoader/LazyLoader';
import * as React from 'react';
import LazyLoadMoreStyles from './LazyLoadMoreStyles';

export interface ILazyLoadMoreTriggerProps {
    loadMoreCallback: () => void;
}

export const LazyLoadMoreTrigger = (props: ILazyLoadMoreTriggerProps) => {

    const [used, setUsed] = React.useState(false);

    const loadMoreTrigger = () => {
        if (!used) {
            setUsed(true);
            props.loadMoreCallback();
        }
        return null;
    }

    const containerRef = React.createRef<HTMLDivElement>();

    return (
        <LazyLoadMoreStyles.LoadMoreRow ref={containerRef}>
            <LazyLoader
                loadingPlaceholder={<AnimatedLoader />}
                containerRef={containerRef}
            >
                {loadMoreTrigger}
            </LazyLoader>
        </LazyLoadMoreStyles.LoadMoreRow>
    );
}
