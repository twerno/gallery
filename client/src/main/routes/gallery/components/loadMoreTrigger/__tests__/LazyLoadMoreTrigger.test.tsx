import '@testing-library/jest-dom/extend-expect';
import { mount } from 'enzyme';
import 'jest-styled-components';
import * as React from 'react';
import { componentBasicTests } from 'test/componentBasicTests';
import { LazyLoadMoreTrigger } from '../LazyLoadMoreTrigger';

describe('LazyLoadMoreTrigger', () => {

    componentBasicTests(
        "LazyLoadMoreTrigger",
        <LazyLoadMoreTrigger loadMoreCallback={() => { }} />
    );

    test("triggers callback", async () => {
        const callbackHandler = jest.fn();

        mount(<LazyLoadMoreTrigger loadMoreCallback={callbackHandler} />);

        expect(callbackHandler).toBeCalled();
    });

});

