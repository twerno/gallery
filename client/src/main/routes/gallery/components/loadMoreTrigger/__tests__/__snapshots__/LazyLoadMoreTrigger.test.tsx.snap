// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyLoadMoreTrigger [LazyLoadMoreTrigger] Component basic tests matches the snapshot 1`] = `
<styled.div>
  <Component
    containerRef={
      Object {
        "current": null,
      }
    }
    loadingPlaceholder={<styled.div />}
  >
    <Component />
  </Component>
</styled.div>
`;
